docs/features/nullable-reference-types.md

```diff
@@ -102,7 +102,15 @@ T t = default; // assigns ?, warning
 ```
 
 ### Conversions
-_Describe valid top-level and variance conversions._
+Conversions can be calculated with ~ considered distinct from ? and !, or with ~ implicitly convertible to ? and !.
+Given `IIn<in T>` and `IOut<out T>`, with ~ distinct:
+- `T!` is a `T~` is a `T?`
+- `IIn<T!>` is a `IIn<T~>` is a `IIn<T?>`
+- `IOut<T?>` is a `IOut<T~>` is a `IOut<T!>`
+Most conversions are considered with ~ implicitly convertible to ? and !.
+The exceptions are method type inference and best type inference. Those cases first try inference with ~ distinct from ? and !,
+and if that fails, inference is considered with ~ convertible to ? and !.
+
```

Comment:
As we discussed, I don't think that is the correct way to perform inference.  Subtyping (during inference) needs to consider that identity conversions exist between all three types in order to get the same result type as during initial binding.  The modification to type inference is in determining the nullable annotations of the result during fixing.

Reply:
It would be great if the speclet could be updated in the PR, as that facilitates the review. Can that be arranged?

---
In reply to: [223449206](https://github.com/dotnet/roslyn/pull/29727#discussion_r223449206) [](ancestors = 223449206)

Reply:
Logged [https://github.com/dotnet/roslyn/issues/30433](https://github.com/dotnet/roslyn/issues/30433).

---
In reply to: [223520316](https://github.com/dotnet/roslyn/pull/29727#discussion_r223520316) [](ancestors = 223520316,223449206)